codice/ddf
# DDF-2311 Fix security filters so they look up the context policy by t…

#### What does this PR do?
-Fixes `WebSSOFilter` and `AuthorizationFilter` so they look up the context policy by the request URI rather than looking it up by the context path or the servlet path
-Delete unused `security-filter-delegate` module

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@codice/security
@bdeining

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@stustison
@pklinef

#### How should this be tested?
Verify that the default context realms, authentication types, required attributes, and whitelisted contexts in the Web Context Policy Manager work as expected.

Install the GeoWebCache app (which works well for testing since the GeoWebCache servlet has a non-empty context path and multiple endpoints). Create some new web context policies for GeoWebCache's endpoints (such as `/geowebcache/rest` or `/geowebcache/demo`) and verify that they work as expected.

#### What are the relevant tickets?
[DDF-2311](https://codice.atlassian.net/browse/DDF-2311)

#### Checklist:
